### PR TITLE
Auto generate regression tests

### PR DESCRIPTION
**Pull request by Droogans**
JIRA: https://jira.rax.io/browse/SURF-704

I can probably continue adding features to this pull request, namely the ability to take the screenshots that get generated during the `watch` feature here in a way that resembles how `visreg` works.

### LGTM's
- [x] Dev LGTM
- [ ] Zoom LGTM

### CODE REVIEW

**Comment by Droogans** [comment]
Now that `yarn start` will start a selenium server, there will need to be an update in the readme/wiki about how to get java installed for this new requirement.

File: test/regression/generateRegressionTests.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import * as util from "../common/util";
+
+let snappit: Snappit;
+let driver: WebDriver;
+
+const grepCommand = 'grep "\\bdata-visreg=" ../source/components/**/*.html';
+const taggedForRegression = child_process.execSync(grepCommand).toString().trim();
+
+const componentExtractor = /\.\.\/source\/components\/(\w+)\/index\.html/gm;
```
Comment (CITguy) [inline test/regression/generateRegressionTests.ts]: This may be easier to grok using `new RegExp()`

File: test/package.json [hunk 2]
```diff
@@ -19,19 +19,23 @@
     "build": "tsc",
     "clean": "npm run clean:visreg && npm run clean:func",
     "clean:func": "find built/functional -mindepth 1 \\( ! -path 'built/functional/*.js.snap' ! -path 'built/functional/*.js.md' \\) -delete",
+    "clean:regression": "find built/regression -mindepth 1 \\( ! -path 'built/regression/*.js.snap' ! -path 'built/regression/*.js.md' \\) -delete",
```
Comment (CITguy) [inline test/package.json]: Please move to script.

File: test/regression/generateRegressionTests.ts (hunk 1 above)
Comment (CITguy) [inline test/regression/generateRegressionTests.ts]: This would be easier to grok with `new RegExp()`

**Comment by CITguy** [comment]
Please make JIRA about asking about SSO/enterprise gh prompt.

**Comment by Droogans** [comment]
https://jira.rax.io/browse/SURF-737 has been made.

**Comment by Droogans** [comment]
@CITguy ready for a re-review.

File: test/built/regression/generateRegressionTests.js.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
+# Snapshot report for `built/regression/generateRegressionTests.js`
```
Comment (CITguy) [inline test/built/regression/generateRegressionTests.js.md]: This works for now, but I think we really need to figure out how to split the results amongst multiple files.

Comment (Droogans) [reply test/built/regression/generateRegressionTests.js.md]: I'll dig into Ava's source and see if there's a way to over-ride the test context's attributes and trick it into thinking that the test is located somewhere that it isn't.

**Pull request closed without merging**